robottelo/ui/role.py

```diff
@@ -38,28 +41,46 @@ def delete(self, name, really=True):
             locators['roles.dropdown'],
         )
 
-    def update(self, name, new_name=None, add_permission=False,
-               resource_type=None, permission_list=None, organization=None):
-        """Update role name/permissions/org."""
-        element = self.search(name)
-
-        if element is None:
-            raise UIError('Could not find role "{0}"'.format(name))
+    def update(self, name, new_name=None, locations=None, organizations=None):
+        """Update role name/location/organization."""
+        self.click(self.search(name))
         if new_name:
-            element.click()
-            if self.wait_until_element(locators['roles.name']):
-                self.field_update('roles.name', new_name)
-        if add_permission:
-            strategy, value = locators['roles.dropdown']
-            self.click((strategy, value % name))
-            self.click(locators['roles.add_permission'])
-            if resource_type:
-                self.select(
-                    locators['roles.select_resource_type'], resource_type)
-                if permission_list:
-                    self.configure_entity(
-                        permission_list, FILTER['role_permission'])
-            if organization:
-                self.click(tab_locators['roles.tab_org'])
-                self.configure_entity(organization, FILTER['role_org'])
+            self.assign_value(locators['roles.name'], new_name)
+        if locations or organizations:
+            self._configure_taxonomies(locations, organizations)
+        self.click(common_locators['submit'])
+
+    def add_permission(self, role_name, resource_type=None,
+                       permission_list=None, override=None,
+                       override_check=None, unlimited=None,
+                       organization=None, location=None):
+        """Add new permission to Role"""
+        self.search(role_name)
+        self.click(locators['roles.dropdown'] % role_name)
+        self.click(locators['roles.add_permission'])
+        if resource_type:
+            self.assign_value(
+                locators['roles.select_resource_type'], resource_type)
+        if permission_list:
+            self.configure_entity(
+                permission_list, FILTER['filter_permission'])
+        # Verify whether 'Override' checkbox is in expected state. Putting it
+        # here as we can do it only during create procedure, not after
+        if override_check is True:
```

Comment:
As stated in the next comment, for non-overridable filters, 'Override' checkbox should be absent on the page. In all other cases it should be present and be in unchecked state